tools/cldr-apps/src/main/java/org/unicode/cldr/web/DataSection.java

```diff
@@ -1850,12 +1851,8 @@ private void ensureComplete(CLDRFile ourSrc, TestResultBundle checkCldr) {
                     // synthesize a new row..
                     String base_xpath_string = podBase + ourSuffix + suff;
 
-                    SurveyToolStatus ststats = SurveyToolStatus.READ_WRITE;
                     PathHeader ph = stf.getPathHeader(base_xpath_string);
-                    if (ph != null) {
-                        ststats = stf.getPathHeader(base_xpath_string).getSurveyToolStatus();
-                    }
-                    if ((ststats == SurveyToolStatus.HIDE || ststats == SurveyToolStatus.DEPRECATED)) {
+                    if (ph == null || ph.shouldHide()) {
```

Comment:
This change assumes that getPathHeader never returns null for a valid path that shouldn't be hidden.

Is my assumption correct?